library/src/main/kotlin/com/connectrpc/impl/ProtocolClient.kt

```diff
@@ -94,7 +96,20 @@ class ProtocolClient(
             val unaryFunc = config.createInterceptorChain()
             val finalRequest = unaryFunc.requestFunction(unaryRequest)
             val cancelable = httpClient.unary(finalRequest) httpClientUnary@{ httpResponse ->
-                val finalResponse = unaryFunc.responseFunction(httpResponse)
+                val finalResponse: HTTPResponse
+                try {
+                    finalResponse = unaryFunc.responseFunction(httpResponse)
+                } catch (ex: Throwable) {
```

Comment:
This was one of the previously uncaught exceptions. It was happening when there was a problem in the wire format of the 5-byte envelopes around stream messages. Instead of being turned into an RPC failure, it was causing an okhttp worker thread to die, which would then cause the conformance client to hang and timeout.